### PR TITLE
ipcache: avoid memory allocations when not running in debug mode

### PR DESCRIPTION
**Pull request by aanm**
Read per commit basis.

This will probably be deprecated once we move out of logrus.

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/8273)

### CODE REVIEW

**Comment by aanm** [comment]
test-me-please

**Comment by ianvernon** [comment]
Checkout from Git failed on Jenkins: https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-Validated/12952/execution/node/7/log/

**Comment by ianvernon** [comment]
test-me-please

File: pkg/ipcache/kvstore.go [hunk 1]
```diff
@@ -217,8 +220,10 @@ restart:
 				goto restart
 			}
 
-			scopedLog := log.WithFields(logrus.Fields{"kvstore-event": event.Typ.String(), "key": event.Key})
-			scopedLog.Debug("Received event")
+			if option.Config.Debug {
+				scopedLog = log.WithFields(logrus.Fields{"kvstore-event": event.Typ.String(), "key": event.Key})
```
Comment (ianvernon) [inline pkg/ipcache/kvstore.go:224]: AFAICT `scopedLog` is never used again - why not just do `log.WithFields(logrus.Fields{"kvstore-event": event.Typ.String(), "key": event.Key}).Debug("Received event")` ?

**Review by ianvernon — APPROVED** [review]

File: pkg/ipcache/kvstore.go (hunk 1 above)
Comment (joestringer) [reply pkg/ipcache/kvstore.go:224]: Looks like it is used again, in the `nil` IP create/modify case below.

**Review by joestringer — APPROVED** [review]
This is kludgy (copy+paste, easy to forget to update an instance), but if we're forming these logs a lot then I guess there's a nice performance win here. Nice finds.

**Pull request merged**